File: app/src/main/java/com/example/shoppingapp/Item.java
package com.example.shoppingapp;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.List;


public class Item implements Parcelable {

  private String name;
  private int icon;
  private String description;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public int getIcon() {
    return icon;
  }

  public void setIcon(int icon) {
    this.icon = icon;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public double getPrice() {
    return price;
  }

  public void setPrice(double price) {
    this.price = price;
  }

  public String getManufactureDetails() {
    return manufactureDetails;
  }

  public void setManufactureDetails(String manufactureDetails) {
    this.manufactureDetails = manufactureDetails;
  }

  private double price;
  private String manufactureDetails;

  @Override
  public int describeContents() {
    return 0;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeString(this.name);
    dest.writeInt(this.icon);
    dest.writeString(this.description);
    dest.writeDouble(this.price);
    dest.writeString(this.manufactureDetails);
  }

  public Item() {
  }

  protected Item(Parcel in) {
    this.name = in.readString();
    this.icon = in.readInt();
    this.description = in.readString();
    this.price = in.readDouble();
    this.manufactureDetails = in.readString();
  }

  public static final Parcelable.Creator<Item> CREATOR = new Parcelable.Creator<Item>() {
    @Override
    public Item createFromParcel(Parcel source) {
      return new Item(source);
    }

    @Override
    public Item[] newArray(int size) {
      return new Item[size];
    }
  };
}

File: app/src/main/java/com/example/shoppingapp/Item.kt
package com.example.shoppingapp

import android.os.Parcel
import android.os.Parcelable

class Item() : Parcelable {

    lateinit var id:String
    lateinit var name: String
    var icon: String? = null
    var description: String? = null
    lateinit var price: String
    var manufacture: String? = null
    get(){
        return if(field=="") "no details added by vendor" else field
    }
    lateinit var stock: String
    lateinit var categoryId:String
    lateinit var vendorId: String

    constructor(parcel: Parcel) : this() {
        id = parcel.readString() as String
        name = parcel.readString() as String
        icon = parcel.readString()
        description = parcel.readString()
        price = parcel.readString() as String
        manufacture = parcel.readString()
        stock = parcel.readString() as String
        categoryId = parcel.readString() as String
        vendorId = parcel.readString() as String
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(id)
        parcel.writeString(name)
        parcel.writeString(icon)
        parcel.writeString(description)
        parcel.writeString(price)
        parcel.writeString(manufacture)
        parcel.writeString(stock)
        parcel.writeString(categoryId)
        parcel.writeString(vendorId)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Item> {
        override fun createFromParcel(parcel: Parcel): Item {
            return Item(parcel)
        }

        override fun newArray(size: Int): Array<Item?> {
            return arrayOfNulls(size)
        }
    }

}